pandas/core/indexes/datetimelike.py

```diff
@@ -794,7 +796,7 @@ def isin(self, values):
 
     def shift(self, n, freq=None):
         """
-        Specialized shift which produces a DatetimeIndex
+        Specialized shift which produces a %(klass)s
```

Comment:
Instead of doing the overriding in the subclass just to fix this name, we can maybe reword the docstring to be clear enough? 
Eg mention it works for Datetime- and TimedeltaIndex?

Reply:
That was the idea in the earlier version.  I'm open to suggestions.